main.go

```diff
@@ -26,6 +26,7 @@ func (i *arrayFlags) Set(value string) error {
 func main() {
 	var flagTest arrayFlags
 	var counter int
+	anyMatch := false
```

Comment:
Don't you already have a variable indicating whether there was any match?

Reply:
Yes but if we have for example one match at the beginning equalComp = true it breaks the for loop print it and continues to read line by line. At the beginning of my big for loop equalComp is set false so if there is no further match it wont work properly

Reply:
I didn't mean equalComp.

Reply:
there is no other var except for counter but i cannot use counter aswell

Reply:
Why not?

Reply:
lets say we go for `go run . -m 0 -e 'dog' < toGrep.txt` even if there is a match counter will be 0 and i would have to check for counter > 0 because if there is not match counter will be 0 as well

Reply:
That's fine.